lib/spack/spack/package.py

```diff
@@ -1926,23 +1926,31 @@ def run_test(self, exe, options, expected, status):
             exe (str): the name of the executable
             options (list of str): list of options to pass to the runner
             expected (list of str): list of expected output strings
-            status (int or None): the expected process status if int or None
-                if the test is expected to succeed
+            status (int or list of int): possible passing status values with
+                0 and None meaning the test should succeed
         """
-        result = 'fail with status {0}'.format(status) if status else 'succeed'
```

Comment:
I think a lot of this code could be cleaned up by putting at the top of the method:

```
if not isinstance(status, list):
    status = [status]
```
And then we can assume it's a list throughout.